.github/workflows/test.yml

```diff
@@ -26,6 +26,6 @@ jobs:
       - name: Run tests
         # Runs a script that creates a PostgreSQL client,
         # populates the client with data, and retrieves data
-        run: go test -cover ./...
+        run: go test -cover -p 1 ./...
```

Comment:
This runs one test at a time to prevent `setupTestDB` from using `ResetPGTestDB` to reset the test database for one test while another is using it. We're up to enough tests now that they interfere with each other in this way. Before this change, only 1/4 test runs succeeded without running into a `Process: ERROR #42P01 relation "lnmux.invoices" does not exist` error, while after this change, 4/4 test runs succeeded.

We may have a few other options on handling this, such as refactoring the tests to explicitly run one at a time from within a test runner that sets up and tears down the database between running tests. However, this doesn't guarantee that tests from one package won't interfere with tests from another package. Another option might to prevent the database resets between tests and instead ensure that each test uses unique invoice IDs, which again may be difficult to coordinate between multiple test files and packages.

This may also extend test run time in the future as we add more tests, which may cause issues with the current test timeout mechanism.

Reply:
Is it right that within a package, we can control this problem by just not marking tests with `t.Parallel()`, but our problem is that both the main package and the persistence package tests execute in parallel?

Reply:
Maybe it is possible to create random databases for each test?

Reply:
> Is it right that within a package, we can control this problem by just not marking tests with `t.Parallel()`, but our problem is that both the main package and the persistence package tests execute in parallel?

Yeah exactly that, like said the `go help` for the test flag: 
_The 'go test' command may run tests for different packages in parallel as well, according to the setting of the -p flag_


> Maybe it is possible to create random databases for each test?

Yeah we could totally do that but why not using the `schema` mechanism already included in the `test.go`?
The only benefit I see is that, we don't have to check if the schema we want to use for a future test package has already been used by another test package.

Reply:
I've changed the code to create a new DB for each tests. This means on a local dev box, the user under which the tests runs needs to be able to create databases (`ALTER ROLE bottle WITH CREATEDB;` or similar). I've run the tests several times in CI (e.g. https://github.com/bottlepay/lnmux/actions/runs/2907465477) and the tests no longer seem to flake/run over each other's DB.

There are other choices of how to do this so I'd love some feedback. There may also be opportunities to remove a few more lines from the code if there's consensus that this is the way to move forward for now.